### ISSUE TITLE
[High Priority :S] TE + TD crash

### ISSUE THREAD

**Issue by Lordmau5** [CoFH/Feedback#287]
![Image](http://puu.sh/guy84/427339626e.jpg)
Pulverizer is set to "Top: Input - Right: Output".
Furnace is set to "Nothing" on each side.

When I now change the side-accesibility for the Furnace, it crashes with this log:
http://puu.sh/guycU/1be12671b6.txt

However, it only crashes if I change a side **OTHER THAN** the one where the Duct-Network is connected to.
In this case, if I change the "Top", it won't crash.
Changing the "Left" to "Input" for example, to get the output of the Pulverizer into the Furnace, crashes it though.

//EDIT: Gotta note that I'm using the latest versions of the mods. CoFHCore, TE and TD.
//EDIT2: Also forgot to mention, that nothing has to be in the Chest for it to crash.

**Comment by Lordmau5** [comment]
Copy of #276, and yes, it is fixed. I can change modes fine :)

**Comment by Lordmau5** [comment]
Reopening, because it just happened again after a while, wow...
Might having a look into it, please?

**Comment by KingLemming** [comment]
Unfortunately, one person reporting a crash that we can't reproduce doesn't make it high priority. :(

**Comment by Carbon-Key** [comment]
I actually can confirm that I got another crash on the dev build after I'd thought the issue was fixed, was messing around in the same world I was using for the screenshot from #276 but this time in RC5-95 and i got http://pastebin.com/dC1Yjq48 when I adjusted the inputs on a machine hooked up to ducts. 

Here is a shot of my setup:
![2015-03-11_02 05 58](https://cloud.githubusercontent.com/assets/10481639/6591252/46db04de-c793-11e4-9c51-b843d9e1fa8e.png)
 
The retrievers are set to pull the items the machines are producing and the other inventories are full of items that are not on the filters/are not cobblestone.  It is a different crash from before but it does occur under the same circumstances.  I'm sorry I wasn't thorough enough initially to catch it.

**Comment by KingLemming** [comment]
Just checking, is this unique to the Fluctuating ducts?

**Comment by Lordmau5** [comment]
Yes, it happens **JUST** with the Fluctuating Itemducts.
I tried it with all the other ones (Normal, Glowstone, Enderium) and those were fine.

**Comment by Carbon-Key** [comment]
I can't double check that right at the moment but I will post as soon as I get back to my computer, just to make sure it is the same for both of us.

**Comment by Lordmau5** [comment]
Isolating it even further.

My friend just tried to reproduce it too, but couldn't.
This is the way he built it:
![Image](http://puu.sh/gwh9f/993c234adc.jpg)
Furnaces set to "Input: Back (Bottom line) - Output: Top (Top line)"

Now, I copied his way of doing things, and it didn't crash me either.

**HOWEVER**:
Upon changing the build do this
![Image2](http://puu.sh/gwhfZ/5be94efb0c.jpg)
it **DOES** crash.
Both on my end *and* his end.

**Comment by Carbon-Key** [comment]
Sorry it took me so long to get back to this. I had to spam the inputs a bit to get it to work (crash), but i did indeed get the crash again with the fluctuating item ducts.  Very interestingly I tried to re-create it with other itemduct types and failed but I DID finally re-create the first crash from issue #276 (with warp item ducts this time).  It looks the same as the earlier one line for line, but in case there is a difference here is the new one: http://pastebin.com/MhC0ZjBz.  Just like before it is stuck in a crash loop on world load.  I am using the same setup from my earlier screenshot in this thread except, to isolate any cross-mod issues, I swapped the barrels for caches and the transfer node for an augmented igneous extruder (and of course the fluctuating ducts for warp ones).  Sorry if any info I've posted is haphazard, just trying to give you as much info as I can.  If you think of something specific I've overlooked that would be helpful please let me know.  I avoided posting this in the original thread ( #276 ) to bring all the information to one place, but if you would like me to post there as well for completeness sake then I can do that.

**Comment by TheJulianJES** [comment]
I can reproduce this issue on my server with the latest stable and alpha versions from all CoFH mods.

**Comment by rwtema** [comment]
Found the issue and fixed in dev.

It happens when fluctuatiing ducts connect to a TE with no accessible slots but still accepts energy. One piece of code says this duct should tick but another incorrectly says it shouldn't. The two fight and it's removed mid-tick and the system crashes.

The temp fix is just to ensure that the fluctuatiing ducts are not placed next to blank sides of the TE4 machines.

**Comment by Carbon-Key** [comment]
That's great news!  I am assuming this does not affect the crash from my previous post however? (Which was involving warp ducts) Would you like me to continue putting information about that here, or in the original issue (#276)?

**Comment by Lordmau5** [comment]
In the meantime I wrote a fix on my own. It still has the *"I'm trying to insert into a inventory, but after I tried I'm just gonna go to the next one"* which causes a buffer at e.g. round robin and stuff, but at least it's not crashing anymore.
Not gonna post it anywhere though, it's just for private use up until you guys fix it properly :P

**Comment by KingLemming** [comment]
@Carbon-Key - what version do you have for the Warp-duct crash?

**Comment by Carbon-Key** [comment]
Ah sorry, I could have sworn I put it somewhere above.  I am still on the dev version you linked me, 1.0.0RC5-95.  Let me double check if it is specific to warp ducts, because originally I got it using fluctuating item ducts.

**Comment by Carbon-Key** [comment]
I have come to the conclusion that I have no idea what is triggering the crash.  I thought it was adjusting the machine I/O but that is no longer making it happen, and the first time I was adjusting the servo settings, not the machine.  I will just try to describe my setup as best I can, it is the exact same (to my knowledge) as what caused the crash last time I tried it.
![2015-03-13_19 35 19](https://cloud.githubusercontent.com/assets/10481639/6649027/2ea82b72-c9b8-11e4-922c-cebe8728f438.png)
![2015-03-13_19 35 34](https://cloud.githubusercontent.com/assets/10481639/6649028/31af102e-c9b8-11e4-8a4c-63e7455e0691.png)

The servo is pulling cobble from the cache with an empty blacklist, the cache is being fed by a fully augmented igneous extruder.  The backs of the machines, redstone furnaces on one side and pulverizers on the other, are being fed cobble.  The tops are set to output all the products of the machines.  All other faces are disabled, the warp duct connections were disabled with a wrench prior to disabling the faces of the machines touching them.  I have removed the integrated servo mechanism augments (automated output) on all machines except the extruder.  The three retrievers are set to a whitelist for the contents of the attached caches (stone, sand and gravel).  The pair of tesseracts are powering the warp ducts, items and liquids are set to blocked on the both.  The machines are powered individually by creative energy cells placed beneath them.  I feel like I'm not giving any new/useful info, but I am really at a loss as to how this crash is happening.  Let me know if I overlooked some aspect of the setup that you are curious about.
